### PR TITLE
Circuit for opcode `STATICCALL`

### LINKED ISSUE #699: Implement opcode `STATICCALL` in Circuit
Please help assign this issue to me if no one is working on this opcode. Thanks.

Related Spec issue - https://github.com/privacy-scaling-explorations/zkevm-specs/issues/252

### PR DESCRIPTION
**Pull request by silathdiir**
Close https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/699

Spec: https://github.com/privacy-scaling-explorations/zkevm-specs/pull/281

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (z2trillion) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:105]: I think you can just do 
```
cb.condition(is_call.expr(), |cb| cb.stack_pop(value.expr()));
```
here, and then you don't need to change the lines after?

**Review by z2trillion — APPROVED** [review]

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (silathdiir) [reply zkevm-circuits/src/evm_circuit/execution/callop.rs:105]: I used `cb.condition(is_call.expr(), |cb| cb.stack_pop(value.expr()))` for this code before. But it seems that not work for me.

I found the `stack_pointer_offset` is just increased in the function `stack_pop` internal (not an expression). This causes the next stack pop/push has wrong `stack_pointer_offset`.

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/zkevm-circuits/src/evm_circuit/util/constraint_builder.rs#L992

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (roynalnaruto) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:128]: These can be replaced with `stack_pop` and `stack_push` right?

**Review by roynalnaruto — COMMENTED** [review]
Did an initial pass. I'm still going through the test cases which will take me a bit more time to review. Overall looks good!

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (z2trillion) [reply zkevm-circuits/src/evm_circuit/execution/callop.rs:105]: I think that is a bug. I made a PR here to turn stack_pointer_offset into an expression so that `self.condition` can be used: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/837.

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (z2trillion) [reply zkevm-circuits/src/evm_circuit/execution/callop.rs:128]: No, so the discussion here: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/801#discussion_r997670440

File: zkevm-circuits/src/evm_circuit/step.rs
Comment (silathdiir) [reply zkevm-circuits/src/evm_circuit/step.rs:83]: I suppose if it could be renamed to `CALL_OP` after merging both `DELEGATECALL` and `CALLCODE` opcodes. WDYT?

File: zkevm-circuits/src/evm_circuit/execution/callop.rs
Comment (roynalnaruto) [reply zkevm-circuits/src/evm_circuit/execution/callop.rs:131]: Okay! Let's wait for #837 to be merged, and then re-run the build for this PR?

**Review by roynalnaruto — APPROVED** [review]

**Pull request merged**
